logprep/processor/generic_resolver/processor.py

```diff
@@ -36,8 +41,77 @@
 class GenericResolver(FieldManager):
     """Resolve values in documents by referencing a mapping list."""
 
+    @define(kw_only=True)
+    class Config(Processor.Config):
+        """ListComparison config"""
+
+        max_cache_entries: Optional[int] = field(
+            validator=validators.optional(validators.instance_of(int)), default=0
+        )
+        """(Optional) Size of cache for results when resolving form a list."""
+        cache_metrics_interval: Optional[int] = field(
+            validator=validators.optional(validators.instance_of(int)), default=1
+        )
+        """(Optional) Cache metrics won't be updated immediately.
+        Instead it's skipped a number of times determined by cache_metrics_interval (default: 1)."""
+
+    @define(kw_only=True)
+    class Metrics(FieldManager.Metrics):
+        """Tracks statistics about the generic resolver"""
+
+        new_results: GaugeMetric = field(
+            factory=lambda: GaugeMetric(
+                description="Number of new values",
+                name="generic_resolver_new_results",
+            )
+        )
+        """Number of new values"""
+
+        cached_results: GaugeMetric = field(
+            factory=lambda: GaugeMetric(
+                description="Number of values resolved from cache",
+                name="generic_resolver_cached_results",
+            )
+        )
+        """Number of resolved values from cache"""
+        num_cache_entries: GaugeMetric = field(
+            factory=lambda: GaugeMetric(
+                description="Number of resolved values in cache",
+                name="generic_resolver_num_cache_entries",
+            )
+        )
+        """Number of values in cache"""
+        cache_load: GaugeMetric = field(
+            factory=lambda: GaugeMetric(
+                description="Relative cache load.",
+                name="generic_resolver_cache_load",
+            )
+        )
+        """Relative cache load."""
+
+    __slots__ = ["_cache_metrics_count"]
+
+    _cache_metrics_count: int
```

Comment:
Maybe we can rename this to something like `_cache_metrics_skip_count` to indicate that it counts the events between the metrics updates.